mlx_graphs/nn/conv/gat_conv.py

```diff
@@ -0,0 +1,98 @@
+from typing import Any
+
+import mlx.core as mx
+import mlx.nn as nn
+import mlx.nn as nn
+
+from mlx_graphs.nn.message_passing import MessagePassing
+from mlx_graphs.utils import glorot_init, scatter_softmax, gather_src_dst
+
+
+class GATConv(MessagePassing):
+    r"""Graph Attention Network convolution layer.
+
+    Args:
+        x_dim (int): size of input node features
+        h_dim (int): size of hidden node embeddings
```

Comment:
these should probably be `input` and `output` dimensions instead of `x` and `h`

Reply:
Yes, as discussed earlier we will not use x and h anymore, so I'll change that too